### PR TITLE
@orta=> Fix for settings button showing sync icon on iPad

### LINKED ISSUE #125: Fix settings icon image during and after syncing
Sometimes the settings icon shows a sync arrow even after the sync has completed.

### PR DESCRIPTION
**Pull request by sarahscott**
Curious about your opinion on this - I'm not quite sure how to test it right now but it works!

Fixes #125

### CODE REVIEW

File: Classes/Sync/ARSync.m [hunk 1]
```diff
@@ -51,7 +55,6 @@ - (void)performSync:(void (^)())completion
 
         // Cleanup the tree so it's recreated next sync
         self.rootOperation = nil;
-        if (completion) completion();
```
Comment (sarahscott) [inline Classes/Sync/ARSync.m]: So the settings button relies on the ARSyncNotificationPlugin, which was running before this completion block was called. The icon would then check sync.isSyncing and receive YES every time.

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

File: docs/CHANGELOG.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 upcoming:
   version: 2.5.0
   notes:
+    - (dev) Fix for main settings button corresponding to sync status [sarah]
```
Comment (sarahscott) [inline docs/CHANGELOG.yml]: Danger is really going to change the changelog game for us all

**Comment by orta** [comment]
Your logic looks good to me. 

OK, so testing this will sound like a bit of a chore, but it's _probably_ worth it. You'd need to create an ARSync that has a stubbed sync tree, you'd need to check the isSyncingStatus before, during and after. Phew.

**Comment by sarahscott** [comment]
I'd like to pair on ARStubbedSync tests because they seem like a rabbit hole and might significantly delay a release if I take them on myself.

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by orta** [comment]
I'm not in town for a month, I'll see if I can make a PR after I make the two eigen releases today

**Comment by orta** [comment]
looking like not tonight

**Comment by orta** [comment]
Looking at this now.

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by sarahscott** [comment]
:green_apple:

**Comment by orta** [comment]
Those are good, but they aren't the things people will forgot in 10 weeks time. These are:

```
ARStubbedSyncProvider.h 
ARStubbedSyncProvider.m 
ARStubbedSyncDelegate.h
ARStubbedSyncDelegate.m
ARStubbedDRBOperation.h
ARStubbedDRBOperation.m
```

They need comments saying what they do, and especially for `ARStubbedDRBOperation.m` - what is different!

**Comment by sarahscott** [comment]
So I think my new comments would make sense to someone who knows how sync works. It'd be kinda cool to have a doc somewhere that explains it in words and diagrams (yours were helpful even if they weren't pretty!).

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by orta** [comment]
Yeah, maybe a blog post on the subject would be a good idea.

**Comment by orta** [comment]
Or depending on how I onboard @mennenia - we can do a video / screenshare of it.

**Pull request merged**